TTHAnalysis/python/plotter/susy-sos/functionsSF.cc

```diff
@@ -4,770 +4,691 @@
+	float eta2     = min(float(2.399), abs(_eta2));
+	float eta3     = min(float(2.399), abs(_eta3));
+	float met      = max(float(50.1) , _met      );
+	float met_corr = max(float(50.1) , _met_corr );
 
-float getMuonSFtracking_16(float pt, float eta, int var = 0){
-	//---pT>10 GeV-------
-	if(pt>10){
-		if     (abs(eta)>0.0  && abs(eta)<=0.20 ) return 0.9800;
-		else if(abs(eta)>0.20 && abs(eta)<=0.40 ) return 0.9862;
-		else if(abs(eta)>0.40 && abs(eta)<=0.60 ) return 0.9872;
-		else if(abs(eta)>0.60 && abs(eta)<=0.80 ) return 0.9845;
-		else if(abs(eta)>0.80 && abs(eta)<=1.00 ) return 0.9847;
-		else if(abs(eta)>1.00 && abs(eta)<=1.20 ) return 0.9801;
-		else if(abs(eta)>1.20 && abs(eta)<=1.40 ) return 0.9825;
-		else if(abs(eta)>1.40 && abs(eta)<=1.60 ) return 0.9754;
-		else if(abs(eta)>1.60 && abs(eta)<=1.80 ) return 0.9860;
-		else if(abs(eta)>1.80 && abs(eta)<=2.00 ) return 0.9810;
-		else if(abs(eta)>2.00 && abs(eta)<=2.20 ) return 0.9815;
-		else if(abs(eta)>2.20 && abs(eta)<=2.40 ) return 0.9687;
-		else return 1.0;
+	// High MET triggers
+	if(met_corr>=200.0){
+		eff_Data	= 0.5 * epsilonInf_Data[_year] * ( TMath::Erf( (met_corr - mean_Data[_year]) / sigma_Data[_year] ) + 1 );
+		eff_MC		= 0.5 * epsilonInf_MC[_year] * ( TMath::Erf( (met_corr - mean_MC[_year]) / sigma_MC[_year] ) + 1 );
+		SF			= eff_Data / eff_MC;
 	}
-
-	// --- pT<10 GeV ---
+	// Low MET triggers
 	else{
-		if     (abs(eta)>0.0  && abs(eta)<=0.20 ) return 0.9968;
-		else if(abs(eta)>0.20 && abs(eta)<=0.40 ) return 0.9975;
-		else if(abs(eta)>0.40 && abs(eta)<=0.60 ) return 0.9979;
-		else if(abs(eta)>0.60 && abs(eta)<=0.80 ) return 0.9978;
-		else if(abs(eta)>0.80 && abs(eta)<=1.00 ) return 0.9980;
-		else if(abs(eta)>1.00 && abs(eta)<=1.20 ) return 0.9971;
-		else if(abs(eta)>1.20 && abs(eta)<=1.40 ) return 0.9961;
-		else if(abs(eta)>1.40 && abs(eta)<=1.60 ) return 0.9954;
-		else if(abs(eta)>1.60 && abs(eta)<=1.80 ) return 0.9955;
-		else if(abs(eta)>1.80 && abs(eta)<=2.00 ) return 0.9941;
-		else if(abs(eta)>2.00 && abs(eta)<=2.20 ) return 0.9925;
-		else if(abs(eta)>2.20 && abs(eta)<=2.40 ) return 0.9866;
-		else return 1.0;
-	}
-}
-
-float getMuonSFrecoToLoose_16(float _pt, float eta, int var = 0){
-	float pt = std::min(float(199.9),_pt);
-	if (pt<25){
-		if(abs(eta)<1.2){
-			if(var>0) return (h_muSF_recoToLoose_lowPt_barrel_16->Eval(pt) + h_muSF_recoToLoose_lowPt_barrel_16->GetErrorYhigh(getBinMuonReco_16(pt)));
-			if(var<0) return (h_muSF_recoToLoose_lowPt_barrel_16->Eval(pt) - h_muSF_recoToLoose_lowPt_barrel_16->GetErrorYlow (getBinMuonReco_16(pt)));
-			return h_muSF_recoToLoose_lowPt_barrel_16->Eval(pt);
+		if (choose_leptons == 12 )		SF = trigger2lSF(_pt1, eta1, _pt2, eta2, met, met_corr, _year);
+		else if (choose_leptons == 13 )	SF = trigger2lSF(_pt1, eta1, _pt3, eta3, met, met_corr, _year);
+		else if (choose_leptons == 23 )	SF = trigger2lSF(_pt2, eta2, _pt3, eta3, met, met_corr, _year);
+		else if (choose_leptons == 123){ 
```

Comment:
Is the following the correct way to treat the case in which we have all 3 leptons being muons in the 3l low MET bin? The previous logic, which I copied here, practically skipped the muon leg of the trigger in the efficiency when that happened but I can't understand why.